server/src/main/java/org/elasticsearch/cluster/metadata/NodesShutdownMetadata.java

```diff
@@ -167,9 +169,8 @@ public int hashCode() {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.field(NODES_FIELD.getPreferredName(), nodes);
-        return builder;
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params ignored) {
+        return ChunkedToXContentHelper.xContentValuesMap(NODES_FIELD.getPreferredName(), nodes);
```

Comment:
Tricky - `XContentBuilder` has too much magic. Looks good tho, just noting that I had to think here.